sbysrc/sby_engine_smtbmc.py

```diff
@@ -186,21 +186,22 @@ def output_callback(line):
             cell_name = match[3]
             prop = task.design_hierarchy.find_property_by_cellname(cell_name)
             prop.status = "FAIL"
-            last_prop = prop
+            last_prop.append(prop)
             return line
 
         match = re.match(r"^## [0-9: ]+ Reached cover statement at (\S+) \((\S+)\) in step \d+.", line)
         if match:
             cell_name = match[2]
             prop = task.design_hierarchy.find_property_by_cellname(cell_name)
             prop.status = "PASS"
-            last_prop = prop
+            last_prop.append(prop)
             return line
 
         match = re.match(r"^## [0-9: ]+ Writing trace to VCD file: (\S+)", line)
         if match and last_prop:
-            last_prop.tracefile = match[1]
-            last_prop = None
+            for p in last_prop:
+                p.tracefile = match[1]
```

Comment:
When my `--keep-going` changes land this will need some further adjustment (which I can also do then), as the same assertion can fail in multiple traces. (I output all failed assertion of the trace whenever a new assertion fails for the first time, marking those that failed before.) Would it be possible to collect all traces here or is there something consuming this data that can handle at most one trace? If there should be only one trace, selecting the first, i.e. shortest trace would be better IMO.

Reply:
The JUnit is the only consumer. Technically the `tracefile="..."` attribute can contain an arbitrary string, so you could put e.g. a space-separated list in there, but I don't think that's that useful and I wouldn't want to make that change without checking in with the customer that ordered that feature. So for now I suggest we just put the first tracefile in there. (`p.tracefile` is initialized to `""` so you can check if it's already set.)